src/core/gosling-component.tsx

```diff
@@ -13,6 +13,7 @@ import { isEqual } from 'lodash-es';
 import * as uuid from 'uuid';
 import { publish } from '../api/pubsub';
 import type { IdTable } from '../api/track-and-view-ids';
+import { preverseZoomStatus } from '../compiler/spec-preprocess';
```

Comment:
Moved under \compiler – a better place than this?

Reply:
Given that this function is only used in `gosling-component.tsx` I think it should be kept in /core? I see you logic though, that it is making manipulations on the spec.

Reply:
Agreed. Would be better to put a file closer to where it is used! Added a util file for this function.